CONTRIBUTING.md

```diff
@@ -10,26 +10,42 @@ Thank you for contributing to Remacs!
 
 ## Writing a great Pull Request (PR)
 
-* Check for Rust compiler warnings, as Travis won't go green if there
+* Follow the existing style for new Rust code. Notes about style
+  can be found in the section on [Remacs style](#remacs-style).
+* Check for Rust compiler warnings, as CI won't go green if there
   are warnings.
 * Format your code with
   [rustfmt-nightly](https://github.com/rust-lang-nursery/rustfmt).
+  You can learn the exact version we use at any time by looking at the
+  .travis.yml file.
 * Add docstrings to your Rust functions `/// This function does ...`
-* _Really_ great PRs include tests. Don't worry if it's not
-  possible to write tests for your code yet (this is common).
+* _Really_ great PRs include tests. It is not always possible to
+  write unit tests for Rust code if it needs to also call C code. We
+  do have a mocking library available to help with this. See
+  `rust_src/src/strings.rs` for examples. Don't worry if it's not
+  possible to write tests for your code yet (this is common). Another
```

Comment:
Could you split this into two or three paragraphs? It's quite big at the moment.

Reply:
I will take another stab at it.